core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -388,6 +393,32 @@ public void updatePrimaryTerm(final long newPrimaryTerm) {
                         e -> failShard("exception during primary term transition", e));
                 primaryTerm = newPrimaryTerm;
                 latch.countDown();
+                // prevent primary relocation handoff while resync is not completed
+                primaryReplicaResyncInProgress = true;
```

Comment:
we discussed possible options. Including follow ups, I will leave a comment on the PR it self so it will be visible